atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPoolImpl.java

```diff
@@ -381,7 +381,9 @@ void runOneTimeStartupChecks() {
             CassandraClientPoolingContainer hostPool = getPreferredHostOrFallBack(req);
 
             try {
-                return runWithPooledResourceRecordingMetrics(hostPool, req.getFunction());
+                V response = runWithPooledResourceRecordingMetrics(hostPool, req.getFunction());
+                blacklist.remove(hostPool.getHost()); // successful request -> can un-blacklist
```

Comment:
will this only do something if we randomly queried this host because all hosts are currently blacklisted?

Reply:
I believe tests show that you un-blacklist hosts even before all hosts are blacklisted.

Reply:
In practice, yes. The `getPreferredHostOrFallBack` will screen out blacklisted hosts unless everyone was blacklisted.

Reply:
@sandorw there's actually an intermediate state in the rolling restart test where both 1 and 2 are on the blacklist. We then pick a random host we haven't already tried, which is only host 1, and then 1 is removed from the blacklist